staging/src/k8s.io/client-go/util/webhook/client.go

```diff
@@ -14,33 +14,37 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package config
+package webhook
 
 import (
 	"context"
-	"encoding/json"
 	"errors"
 	"fmt"
 	"net"
 	"net/url"
 
-	lru "github.com/hashicorp/golang-lru"
-	admissionv1beta1 "k8s.io/api/admission/v1beta1"
-	"k8s.io/api/admissionregistration/v1beta1"
+	"github.com/hashicorp/golang-lru"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
-	webhookerrors "k8s.io/apiserver/pkg/admission/plugin/webhook/errors"
 	"k8s.io/client-go/rest"
 )
 
 const (
 	defaultCacheSize = 200
 )
 
-var (
-	ErrNeedServiceOrURL = errors.New("webhook configuration must have either service or URL")
-)
+// ClientConfigInterface is the interface to access webhook client config to share webhook connect code between
+// admission webhooks and CRD conversion webhooks.
+type ClientConfigInterface interface {
```

Comment:
Why not an options struct instead of making folks implement an interface?

Reply:
+1. Matches our Go style much better.

Reply:
Interface was your suggestion Daniel :)

By option struct you mean an struct that has all of these values and constructor for any API that has those fields? I don't see the benefit of that only duplicating data.

Reply:
I made it a struct.